### PR TITLE
solution task a-tiny-JS-world-OOP

### PR DESCRIPTION
**Pull request by VikTorNik**
### tiny-JS-world--OOP

[Demo](https://viktornik.github.io/a-tiny-JS-world--OOP/) |
[Code base](https://github.com/VikTorNik/a-tiny-JS-world--OOP)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by VikTorNik** [comment]
The basic requirements of the task have been fulfilled.
Maybe there are flaws. I'm asking for a review.
Sincerely, Viktor

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+FICTIONALWORLD.map(unitWorld => {
+   let storyYourself = [];
+   unitWorld.hasOwnProperty('voice') ? storyYourself.push(`${unitWorld.voice}`) : false;
+   unitWorld.hasOwnProperty('name') ? storyYourself.push(`${DESCRIPTION.name} ${unitWorld.name}.`) : false;
+   unitWorld.hasOwnProperty('gender') ? storyYourself.push(`${DESCRIPTION.gender} ${unitWorld.gender}.`) : false;
+   unitWorld.hasOwnProperty('person') ? storyYourself.push(`${DESCRIPTION.person} ${unitWorld.person}.`) : false;
+   unitWorld.hasOwnProperty('hands') ? storyYourself.push(`У мене ${unitWorld.hands} ${DESCRIPTION.hands}.`) : false;
+   unitWorld.hasOwnProperty('legs') ? storyYourself.push(`У мене ${unitWorld.legs} ${DESCRIPTION.legs}.`) : false;
+   print(storyYourself.join(' '));
+});
```
Comment (OleksiyRudenko) [inline submissions/VikTorNik/a-tiny-JS-world--OOP/index.js:89]: Let every class handle properties it owns.
This is covered by OOP theory materials. You wont' need any conditionals if classes are designed properly.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@VikTorNik too far from OOP principles.
Revisit the learning materials.

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js (hunk 1 above)
Comment (VikTorNik) [reply submissions/VikTorNik/a-tiny-JS-world--OOP/index.js]: Thank you. I fixed everything.

**Comment by VikTorNik** [comment]
I fixed bug. 
Please, review.

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+   new Man('Petr', 'Hello!'),
+   new Man('Mark', 'Hey!'),
+   new Woman('Natali', 'Hello!'),
+   new Woman('Lesia', 'Hey!'),
+   new Dog('Bublyk', 'male'),
+   new Cat('Murchik', 'male'),
+];
+
+FICTIONALWORLD.map(unitWorld => {
```
Comment (OleksiyRudenko) [inline submissions/VikTorNik/a-tiny-JS-world--OOP/index.js:92]: (1) `map` semantics assumes that input array's elements will be converted into something else and stored in output. Doesn't look it is the case here. Why not `forEach`?

(2) Does fictional world contains a set of "unitWorld"s?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@VikTorNik quite a decent job.
Let's polish it

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js (hunk 2 above)
Comment (VikTorNik) [reply submissions/VikTorNik/a-tiny-JS-world--OOP/index.js]: Yes. forEach more appropriate.
The name of the fixes.
Thank you.

**Comment by VikTorNik** [comment]
I fixed bug.
Please, review.

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   }
+}
+
+class Dog extends Animal {
+   constructor(name, gender) {
+      super('dog', name, gender, 'Woof!');
+   }
+}
+
+class BasePersonHuman extends BasePersonage {
```
Comment (OleksiyRudenko) [inline submissions/VikTorNik/a-tiny-JS-world--OOP/index.js:54]: Why not just `Human`? KISS

File: submissions/VikTorNik/a-tiny-JS-world--OOP/index.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { print } from './js/lib.js';
+
+class BasePersonage {
```
Comment (OleksiyRudenko) [inline submissions/VikTorNik/a-tiny-JS-world--OOP/index.js:3]: WHy not just `Personage`? You will never know if a class you design wouldn't become a base class for a new class.

**Review by OleksiyRudenko — APPROVED** [review]
@VikTorNik well done!

**Pull request merged**